src/renderer/services/docker/path.ts

```diff
@@ -0,0 +1,87 @@
+import logger from "electron-log";
+import database from "../db/api/database";
+
+import {runCmdAsync} from "../utils/cmd";
+import {Command} from "./command";
+import {extractDockerVersion} from "./utils";
+
+export class DockerPath {
+    private defaultPaths = [
+        "docker",
+        "/usr/bin/docker",
+        "/usr/local/bin/docker",
+        "C:\\Program Files\\Docker Toolbox\\docker.exe",
+        "C:\\Program Files\\Docker\\docker.exe",
+        "C:\\Program Files\\Docker\\Docker\\resources\\bin\\docker.exe",
+    ];
+
+    private path: string;
+
+    public static async isValidPath(path: string): Promise<boolean> {
+        try {
+            const result = await runCmdAsync(await Command.version(path));
+            return !!extractDockerVersion(result.stdout);
+        } catch (e) {
+            logger.error(`Error while checking if Docker path is valid: ${e.message}`);
+            return false;
+        }
+    }
+
+    public async getPath(): Promise<string> {
+        if (this.path) {
+            return this.path;
+        }
+
+        const loadedPath = await this.loadPath();
+        if (loadedPath) {
+            return loadedPath;
+        }
+
+        throw new Error("Docker path is not set.");
+    }
+
+    public async findPath(): Promise<string|null> {
+        for (let i = 0; i < this.defaultPaths.length; i ++) {
```

Comment:
forEach? :disappointed:

Reply:
Can't get out of forEach loop :P

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find :stuck_out_tongue: